addons/mail/models/mail_thread.py

```diff
@@ -1745,7 +1745,11 @@ def _message_post_process_attachments(self, attachments, attachment_ids, message
                     continue
                 if isinstance(content, str):
                     encoding = info and info.get('encoding')
-                    content = content.encode(encoding or 'utf-8')
+                    try:
+                        content = content.encode(encoding or 'utf-8')
+                    except Exception as e:
+                        _logger.warning("Failed to encode the content of the attachment with error: %s", str(e))
```

Comment:
Do we need the try/except actually ? is it interesting to have a log ?

cc @Julien00859 because encoding in emails, attachments, security, attachments, python3.14

Reply:
In the particular case of my ticket, I could encode without raising exception if I use utf-8 rather than us-ascii (us-ascii is defined by the 'charset' tag in the email). Maybe it is better to try the `encoding` first, and fallback on 'utf-8' ?

Reply:
Encoding given by email does not match the encoding of the file ? Or is it another issue ?

Reply:
That's it, the xml file is provided in the ticket: https://www.odoo.com/web#id=3144519&cids=1&menu_id=4720&action=333&active_id=49&model=project.task&view_type=form
In this case, I think the enterprise generating the attachment is not making sense (there is no tag mentioning the encoding of the file, and the streetName "Chauss��e" has unknown characters).